build.gradle

```diff
@@ -297,10 +297,10 @@ platform {
 	bundle 'asm:asm-analysis:3.3.1'
 
 	// ivy with at least 2.4 because of bug in IvySettings
-	bundle 'org.apache.ivy:ivy:2.4.0'
+	bundle 'org.apache.ivy:ivy:2.5.2'
 
 	// web stuff
-	bundle 'commons-fileupload:commons-fileupload:1.3.1'
+	bundle 'commons-fileupload:commons-fileupload:1.5'
```

Comment:
Generally, even better than updating is probably removing stuff from the platform that is no longer needed. 
These dependencies listed here under "web stuff" are candidates for removal if they are in fact no longer used. There were some hale server components that were removed some time ago, that might have used these.
Quite certain that at least `openid4java`, `recaptcha4j` and anything related to Wicket can be removed.

Reply:
I'll remove them all 3:
```
bundle 'commons-fileupload:commons-fileupload:1.5'
bundle 'org.openid4java:openid4java:0.9.8'
bundle 'net.tanesha.recaptcha4j:recaptcha4j:0.0.8'
```

As for wicket I found that is pretty used, there is an entire shared module
```
from('modules/shared/wicket.gradle') {
	wicket(wicketVersion, [
		'wicket-core',
		'wicket-request',
		'wicket-util',
		'wicket-spring',
		'wicket-extensions'
	])
}
```
`bundle 'org.codehaus.groovy.modules.http-builder:http-builder:0.7.1'`

Can you confirm the removal please?

Reply:
On `openid4java`, `recaptcha4j` and everything Wicket I'm sure.

I can't tell you for sure for the others, ultimately you need to test the changes with hale studio. As a first hint you could try to see if the bundles appear in a built version of hale studio or in the Tests or Infocenter products.

Reply:
I would need more details about what you mean with
"the bundles appear in a built version of hale studio or in the Tests or Infocenter products".

for example if I remove the [line related to bundle 'org.openid4java:openid4java:0.9.8'](https://github.com/halestudio/hale-platform/blob/bd60c74073c87613b5ff6b35296e808dd59d4c8e/build.gradle#L305)
I get an error for the [bundleFeatures](https://github.com/halestudio/hale-platform/blob/bd60c74073c87613b5ff6b35296e808dd59d4c8e/build.gradle#L939) and I see that the org.openid4java is used in [hale studio here ](https://github.com/halestudio/hale/blob/7e35a00782be84bd85ccc71d007061ee31a30764/build/bundleArtifactMap.json#L222)

Reply:
> I would need more details about what you mean with
"the bundles appear in a built version of hale studio or in the Tests or Infocenter products".

The HALE product is feature based so it is not easy to determine directly from the product file, other options are:

- inspect the list of bundles shipped with a built version of hale studio (create or download build, check `plugins` folder)
- another way is possibly creating a run configuration from the product file and editing the run configuration to check for included bundles (that also allows for easy filtering)

The products Tests and Infocenter are plugin based, so they can be easily checked if a plugin is included or not:

- https://github.com/halestudio/hale/blob/master/common/plugins/eu.esdihumboldt.hale.common.test/Tests.product
- https://github.com/halestudio/hale/blob/master/doc/plugins/eu.esdihumboldt.hale.doc.application/Infocenter.product

> I see that the org.openid4java is used in hale studio here (https://github.com/halestudio/hale/blob/7e35a00782be84bd85ccc71d007061ee31a30764/build/bundleArtifactMap.json#L222)

This Json file supports in mapping from bundle names to Maven artifacts. Something being present there does not mean it is being used, only that it was used at some point.